test/eigen.jl

```diff
@@ -75,24 +75,31 @@ using StaticArrays, Test, LinearAlgebra
         @test vals::SVector ≈ sort(m_d)
         @test eigvals(m_c) ≈ sort(m_d)
         @test eigvals(Hermitian(m_c)) ≈ sort(m_d)
+    end
 
-        # issue #523
-        for (i, j) in ((1, 2), (2, 1)), uplo in (:U, :L)
-            A = SMatrix{2,2,Float64}((i, 0, 0, j))
-            E = eigen(Symmetric(A, uplo))
-            @test eigvecs(E) * SDiagonal(eigvals(E)) * eigvecs(E)' ≈ A
-        end
-
-        m1_a = randn(2,2)
-        m1_a = m1_a*m1_a'
-        m1 = SMatrix{2,2}(m1_a)
-        m2_a = randn(2,2)
-        m2_a = m2_a*m2_a'
-        m2 = SMatrix{2,2}(m2_a)
-        @test (@inferred_maybe_allow SVector{2,ComplexF64} eigvals(m1, m2)) ≈ eigvals(m1_a, m2_a)
-        @test (@inferred_maybe_allow SVector{2,ComplexF64} eigvals(Symmetric(m1), Symmetric(m2))) ≈ eigvals(Symmetric(m1_a), Symmetric(m2_a))
+    # issue #523, #694
+    zero = 0.0
+    smallest_non_zero = nextfloat(zero)
+    smallest_normal = floatmin(zero)
+    largest_subnormal = prevfloat(smallest_normal)
+    epsilon = eps(1.0)
+    one_p_epsilon = 1.0 + epsilon
+    degenerate = (zero, -1, 1, smallest_non_zero, smallest_normal, largest_subnormal, epsilon, one_p_epsilon, -one_p_epsilon)
+    @testset "2×2 degenerate cases" for (i, j, k) in zip(degenerate,degenerate,degenerate), uplo in (:U, :L)
```

Comment:
I don't think this `zip` does what you want because `i,j,k` are always the same? Though I'm not sure what you want!

Perhaps you meant to use `for (i, j, k) in Iterators.product(degenerate,degenerate,degenerate)`?

Reply:
You're totally right, that's what I meant